core/trino-main/src/main/java/io/trino/execution/executor/RunningSplitInfo.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.executor;
+
+import com.google.common.collect.ComparisonChain;
+import io.trino.execution.TaskId;
+
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A class representing a split that is running on the TaskRunner.
+ * It has a Thread object that gets assigned while assigning the split
+ * to the taskRunner. However, when the TaskRunner moves to a different split,
+ * the thread stored here will not remain assigned to this split anymore.
+ */
+public class RunningSplitInfo
+        implements Comparable<RunningSplitInfo>
+{
+    private final long startTime;
+    private final String threadId;
+    private final Thread thread;
+    private boolean printed;
```

Comment:
I'd move this to the bottom of the list... I like to have the immutable state together especially when it is directly assigned from the constructor args